themes/digital.gov/layouts/partials/core/usa-in-page-nav.html

```diff
@@ -0,0 +1,18 @@
+<!-- partials/core/usa-in-page-nav.html -->
+<div class="usa-in-page-nav-container">
+  <!-- 
+    data-root-margin sets the observable "window" for the element, details here https://designsystem.digital.gov/components/in-page-navigation/#using-the-in-page-navigation-component-2
```

Comment:
I would add some comments about what this component does like how you did for #7748.

You could also move the long comment on line 4 into it to describe the settings used.